jbrowse/src/org/labkey/jbrowse/JBrowseLuceneSearch.java

```diff
@@ -65,6 +70,10 @@ public class JBrowseLuceneSearch
     private final String[] specialStartPatterns = {"*:* -", "+", "-"};
     private static final String ALL_DOCS = "all";
     private static final String GENOMIC_POSITION = "genomicPosition";
+    private static final int maxCachedQueries = 25;
+    private static final long maxRamBytesUsed = 250 * 1024 * 1024L;
+
+    private static final ConcurrentHashMap<String, QueryCache> queryCaches = new ConcurrentHashMap<>();
```

Comment:
Can you try something like this:

private final Cache<String, QueryCache> _cache;

and then create that in the constructor:

```
_cache = CacheManager.getStringKeyCache(1000, CacheManager.UNLIMITED, "JBrowseLuceneSearchCache");

```